### ISSUE TITLE
Improve reliability of long requests

### ISSUE THREAD

**Issue by smithara** [ESA-VirES/VirES-Python-Client#26]
Sometimes a long request (probably chunked) can fail because the server is down for redeployment, or the connection is otherwise lost. This causes the whole chunked request to fail with a ConnectionRefusedError. It would be good to handle this case by waiting and resubmitting the job and continuing.

**Comment by pacesm** [comment]
FYI, I case of asynchronous requests it might be a bit more complicated.

When a connection fails during posting of a new job the job is not created. But when the connection failure happens while polling the job status the job still exists on the server. The asynchronous jobs are never lost and they are always executed until they either successfully finish or fail with an error.

In case of a server restart, the running jobs are interrupted but they get automatically reloaded when the server is up and running again.

In other words, once an asynchronous job was accepted by the server the client should assume it still exists even if its status cannot be temporally retrieved.
